### ISSUE TITLE
USWDS - Bug: USA Tooltip clips when tabbed into element and inside of accordion files

### ISSUE THREAD

**Issue by inregards2pluto** [uswds/uswds#5950]
### Describe the bug

We applied the `usa-tooltip` class to a `<span>` element containing a glossary term so users could hover over the word and see the associated definition. The `title` attribute on the original word contains the text we want to appear in the tooltip. The `usa-tooltip` component largely works as expected, but we discovered two situations where clipping occurs:

1) When the tooltip component is used inside of an accordion element and
2) when someone tabs into an element rather than hovers over the element.

We think we can get around the accordion issue by setting the "overflow" CSS property on the accordion element to "visible". However, the clipping when navigating using a keyboard (specifically the <kbd>Tab</kbd> key) is more problematic.

We also noticed that, when using the tab key to navigate between elements with the `usa-tooltip` class, it does not respect values supplied via the `data-position` attribute. E.g. if I add the `data-position="top"` attribute, this is respected when hovering with a mouse cursor. It gets completely ignored when focusing the element via the <kbd>Tab</kbd> key and tooltips appear in seemingly random positions.

### Steps to reproduce the bug

1. Go to web page containing `<span>` elements that have the `usa-tooltip` class and the `data-position="top"` property applied to them. Have at least one element occurring close to the edge of the screen.
2. Hover over elements using mouse cursor to confirm expected behavior.
3. Press the <kb>Tab</kbd> key to navigate into the same elements.
4. User should see data-position varies from when hovering over same element with a mouse
5. User should see tooltip clipped by edge of screen when focusing an element close to edge of the screen via <kb>Tab</kbd>.

### Expected Behavior

I expected that, when I press the <kbd>Tab</kbd> key to focus an element with a tooltip, a tooltip appears that

1) respects any `data-position` attribute assigned to it and
2) is "edge-aware" and will re-position and/or restructure itself to avoid being clipped by the screen.

### Screenshots

![image](https://github.com/uswds/uswds/assets/91487237/497819f3-1c55-411c-bc84-73fd622a3d3b)
![tooltip_clipped_in_accordion](https://github.com/uswds/uswds/assets/91487237/b8ac2caa-6c91-43e3-89ca-234ebcb754dd)

### System setup

- USWDS version: 3.7.0
- Device: MacBook Pro 2023
- Operating system: Sonoma 14.5
- Browser and version: Google Chrome Version 125.0.6422.142 (Official Build) (arm64)

### Additional context

This is being implemented on a Drupal website in conjunction with the [Gesso USWDS starter theme, v 5.0.9](https://www.drupal.org/project/guswds). We are testing this feature on local and develop only, so we do not have a public website we can currently point to for demonstrating this behavior.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by inregards2pluto** [comment]
I'm testing this further and it seems like it might possibly be related to when the window is scrolling and MAYBE events firing out of order? When I am tabbing through terms in a page and the the window scrolls up or down to center the focused element, that's when the tooltip tends to get messed up. However, when I tab between terms that are close enough to each other the window doesn't need to scroll up or down to center the focused element, THEN the behavior acts as expected.

So perhaps the scrolling is messing up some of the calculations used to determine whether the element is in the viewport?

**Comment by inregards2pluto** [comment]
I added a debounce function to the `showToolTip()` function in the `index.js` file in the `usa-tooltip` package and it seems to resolve the issue. However, I had to set the debounce to 500 ms for it to reliably behave as expected. There is a notable lag between the focusin/mouseover event and the tooltip showing, especially compared to the speed it appeared at prior to adding the debounce.

I tried to instead modify the `constant tooltip = behavior()` so that, if it detected that scrolling was happening, it would not run `showToolTip` until after it stopped moving. That way, the tooltip still appears with the normal speed if no scrolling is needed, but pauses until scrolling is complete if it detects that is happening. But it looks like there's a difference in how the browser determines a scroll event that a user is manually performing versus a scroll event done automatically by the browser to center a focused element. As a result, I wasn't able to get it quite working right.

Due to time limitations, we are going to use the below patch that uses the debounce on the `showTooltip()` function. We will keep an eye on this ticket for a more robust fix and remove the patch if/when it is merged into the repo.

Patch:
[@uswds+uswds+3.7.0.patch](https://github.com/user-attachments/files/15874182/%40uswds%2Buswds%2B3.7.0.patch)

**Comment by jonathanbobel** [comment]
I was not able to reproduce this bug. In testing the tooltip within an accordion using both a mouse hover and keyboard tab the content was still visible outside of the accordion's border.

I am wondering if, in the example, the accordion component does not have the native `overflow: visible` style applied to it (or possibly a parent div) that is cutting off the content with `overflow: hidden` or something similar. Without testing the actual implementation mentioned in the issue I can not verify this specific scenario. 

Additionally, in the [guidance for the tooltip component](https://designsystem.digital.gov/components/tooltip/) recommends keeping the content brief:
_**Brief descriptions.** Tooltips perform best with succinct helper text._

Not that that would prevent this issue from happening in this specific scenario, but something to consider as you're crafting content for your application.

**Comment by chandracarney** [comment]
Thanks for looking into this, @jonathanbobel, I'll close this issue for now based on your findings.

If anyone is still experiencing this issue, please feel free to re-open this and continue discussion. Thanks, all!
